plutus-core/plutus-core/src/PlutusCore/Builtin/Polymorphism.hs

```diff
@@ -256,47 +250,3 @@ instance TypeError NoConstraintsErrMsg => Semigroup (Opaque val rep) where
 
 instance TypeError NoConstraintsErrMsg => Monoid (Opaque val rep) where
     mempty = underTypeError
-
-type NoRegularlyAppliedHkVarsMsg =
-    'Text "Built-in functions are not allowed to have higher-kinded type variables" ':<>:
-    'Text " applied via regular type application" ':$$:
-    'Text "To fix this error instantiate all higher-kinded type variables"
-
-instance TypeError NoRegularlyAppliedHkVarsMsg => Functor (Opaque val) where
```

Comment:
It's nice to remove these specific instances. I assume you can do something similar to the `TypeError NoConstraintsErrMsg => Eq (Opaque val rep)` instance and similar ones?

Reply:
No, unfortunately. We can barf on any applied type variable, but we can't barf on any variable appearing in the argument position as it's perfectly fine to have an argument whose type is a type variable (as in `IfThenElse`). It's really about not allowing any constraints and for that the best we can do is hardcode some of them.